tests/apollo/util/bft.py

```diff
@@ -837,15 +837,16 @@ async def wait_for_fetching_state(self, replica_id):
         Returns the current source replica for state transfer.
         """
         with log.start_action(action_type="wait_for_fetching_state", replica=replica_id) as action:
-            with trio.fail_after(10): # seconds
-                while True:
-                    with trio.move_on_after(.5): # seconds
-                        is_fetching = await self.is_fetching(replica_id)
-                        source_replica_id = await self.source_replica(replica_id)
-                        if is_fetching:
-                            action.add_success_fields(source_replica_id=source_replica_id)
-                            return source_replica_id
-                        await trio.sleep(0.1)
+            async def wait_for_fetching_state_predicate():
```

Comment:
I'd suggest simply `is_fetching_state()`, instead of `wait_for_fetching_state_predicate()`